ciao-controller/cnci.go

```diff
@@ -0,0 +1,588 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/01org/ciao/ciao-controller/types"
+	"github.com/01org/ciao/payloads"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+)
+
+// CNCIState represents specific allowed state strings
+type CNCIState string
+
+var (
+	exited CNCIState = payloads.Exited
+	active CNCIState = payloads.Running
+	failed CNCIState = payloads.ExitFailed
+)
+
+type event string
+
+var (
+	added        event = "concentrator added"
+	startFailure event = "cnci start failure"
+	removed      event = "concentrator removed"
+)
+
+var cnciEventTimeout = (5 * time.Minute)
```

Comment:
@rbradford true - I was worried on a busy/large cluster it might take some time to hear back from launcher.